backend/src/migrations/database_optimization.py

```
@@ -0,0 +1,548 @@
+
+    def _execute_index_commands(self, index_commands: List[str]) -> List[str]:
+        """Execute index creation commands and return successful ones"""
+        successful_indexes = []
+
+        for command in index_commands:
+            try:
+                self.db.execute(text(command))
+                self.db.commit()
+                successful_indexes.append(command)
+                logger.info(f"Successfully created index: {command}")
+            except Exception as e:
```

Ruff (0.14.0)

Do not catch blind exception: `Exception`

(BLE001)